drivers/azure/azureutil/securityrules.go

```diff
@@ -0,0 +1,319 @@
+package azureutil
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2019-12-01/network"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/rancher/machine/drivers/azure/logutil"
+	"github.com/rancher/machine/drivers/driverutil"
+	"github.com/rancher/machine/libmachine/log"
+)
+
+const (
+	sshPriority      = 100
+	sshPolicyName    = "SSHAllowAny"
+	dockerPriority   = 300 // NOTE(aiyengar2): must be greater than sshPriority
+	dockerPolicyName = "DockerAllowAny"
+)
+
+// ConfigureSecurityRules attempts to reconcile an already created NSG's rules with the rules requested
+// Priorities set for the rules must be unique and cannot interfere with existing rules
+func ConfigureSecurityRules(rules *[]network.SecurityRule, sshPort, dockerPort int, extraPorts []string) (*[]network.SecurityRule, error) {
+	var err error
+	if rules, err = addSSHRule(rules, sshPort); err != nil {
```

Comment:
switch these both to just pass the pointer and append and return only the err, no need to copy and return a new list

Reply:
sounds good, added to next commit!